### DISCUSSION TITLE
Help with importing meshes

### DISCUSSION THREAD

**Discussion by Omniscye** [nesrak1/UABEA#394] [General]
Hello, stupid question.
How does this work? I use Unity all the time and have used many tools but all i have is export raw data and that makes a .dat file. I can't 
I see textures plugin actually in Github but how about meshes?

**Comment by woctezuma** [comment]
Personally, I have used https://github.com/AssetRipper/AssetRipper which is linked in the README.

**Reply by Omniscye** [reply]
Does that allow importing back into the bundles?  I tried raw data but model just goes "Poof" after.

**Reply by Omniscye** [reply]
> Personally, I have used https://github.com/AssetRipper/AssetRipper which is linked in the README.

Ah nvm yes I have used this in the past.  But this doesn't help me with adding the edited files into the bundle.  But thanks

**Reply by woctezuma** [reply]
Sorry, I have only used the extractor, I have never edited the files back.

**Comment by nesrak1** [comment]
> and have used many tools but all i have is export raw data and that makes a .dat file. 

UABEA is good right now for looking at the raw dumps and editing the raw dumps, but UABE and UABEA as well have never been great at asset export. There is mesh stuff in progress but nothing right now. As @woctezuma says, you can use AssetRipper or the AssetStudio fork (https://github.com/aelurum/AssetStudio) if you want meshes. Mesh import is possible (really anything is possible), just not fun to do since it requires you to build a unity project with the mesh in it with the same engine version and then copy the mesh dumps and (sometimes resS data) to get it in the game to mod.

**Comment by Omniscye** [comment]
Hello and thanks for the response.  Yes I managed to export and then use the same unity version to export and extract raw data of the mesh I wanted ( after editing in blender ).  I'm used to that at least haha. The issue is after I import Raw it corrupts or something. It's very odd! Lol. The sizing is matched in blender to and again same unity version. Also if I tested finding the new mesh I imported into the asset bundle in Asset Studio, it doesn't see it but UBAE does.  Just very odd. Haven't got meshes to actually work.

AssetStudio mod browses better to look through etc and gets ID's but doesn't import so I use UBAE Avia or whatever its called but I think it meshes up imports of raw data lol. Textures work tho

**Reply by nesrak1** [reply]
If you export and import dump in json format instead of raw dump, you get a little bit better error checking on fields which might help. Have you checked if the asset is using a resS?

**Reply by Omniscye** [reply]
Hmmm I haven't ever tried opening a res pack. I see them yes not sure if they are using it.  So I tested again ( resized in blender ). And in-game the model is just gone so ya something is def up haha!   Even music for some reason I can't get working.  I've done a ton of modding to but again, have been in Unreal for over a year now... Ill peek in the res files and see what's happening but damn Unity is annoying compared to UE. So many resources for UE!!!!   Thanks for the reply!   The game is Liar's bar btw.   Great game just like doing cosmetic work. Textures are easy obv but yeah. It's most likely the res files!

**Reply by nesrak1** [reply]
You can't open a resS file in uabea directly. When you export dump, there is usually an m_StreamData field that has a path to a resS file which contains vertex data and the like. If you're not changing this to your new resS, the mesh will definitely not show up.

**Reply by Omniscye** [reply]
Ya thanks.  I know UABEA can't open them as they are raw data. I have streaming assets etc as I dumped everything but nothing that looks like what I need. I don't think doing models like this is worth it, plus even if I get it working - Game patches and boom, gotta redo a ton.  Thanks for the help!  Ima stop doing Unity for the time being as this is just wack lmao

**Reply by nesrak1** [reply]
This is a pc game, so why not just use a mod loader? That makes mesh loading much easier as you can directly load from a bundle.

**Reply by Omniscye** [reply]
Because I haven't used a mod loader before like that. BepinX sure, but that was for my own mods via an SDK that I Knew would work.... It was for Lethal Company, but  then I swapped to Unreal.   I don't see how a mod loader would help me. The issue isn't loading the mods it's making the files work with the game in general. Until I I figure out why the assets aren't being imported properly I have no way of knowing why.  If it's a package/bundle issue, the mod loader would just load a bad package making it moot.

There is a reason this game has limited mods atm. I tend to strive towards harder games as it makes my brain happy.  Any ways, I will continue my investigation, thanks.

**Reply by nesrak1** [reply]
Importing meshes via raw import and resS is _significantly_ more complicated than just making a bundle in unity editor and doing a simple AssetBundle.LoadFile from c#. People who tend to use this for mesh import generally have some reason against c# mods such as a mobile or console game.
